sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_shared/models.py

```diff
@@ -173,19 +198,82 @@ def __init__(self, user_id, **kwargs):
         if self.raw_id is None:
             self.raw_id = _microsoft_teams_user_raw_id(self)
 
+    def __eq__(self, other):
+        return self.raw_id == _microsoft_teams_user_raw_id(other)
+
 
 def _microsoft_teams_user_raw_id(identifier: MicrosoftTeamsUserIdentifier) -> str:
     user_id = identifier.properties['user_id']
     if identifier.properties['is_anonymous']:
-        return '8:teamsvisitor:{}'.format(user_id)
+        return f'{TEAMS_USER_ANONYMOUS_PREFIX}{user_id}'
     cloud = identifier.properties['cloud']
     if cloud == CommunicationCloudEnvironment.DOD:
-        return '8:dod:{}'.format(user_id)
+        return f'{TEAMS_USER_DOD_CLOUD_PREFIX}{user_id}'
     elif cloud == CommunicationCloudEnvironment.GCCH:
-        return '8:gcch:{}'.format(user_id)
+        return f'{TEAMS_USER_GCCH_CLOUD_PREFIX}{user_id}'
     elif cloud == CommunicationCloudEnvironment.PUBLIC:
-        return '8:orgid:{}'.format(user_id)
-    return '8:orgid:{}'.format(user_id)
+        return f'{TEAMS_USER_PUBLIC_CLOUD_PREFIX}{user_id}'
+    return f'{TEAMS_USER_PUBLIC_CLOUD_PREFIX}{user_id}'
+
+
+MicrosoftBotProperties = TypedDict(
+    'MicrosoftBotProperties',
+    bot_id=str,
+    is_resource_account_configured=bool,
+    cloud=Union[CommunicationCloudEnvironment, str]
+)
+
+
+class MicrosoftBotIdentifier(object):
+    """Represents an identifier for a Microsoft bot.
+
+    :ivar str raw_id: Optional raw ID of the identifier.
+    :ivar kind: The type of identifier.
+    :vartype kind: str or CommunicationIdentifierKind
+    :ivar MicrosoftBotProperties: The properties of the identifier.
+     The keys in this mapping include:
+        - `bot_id`(str): The id of the Microsoft bot.
+        - `is_resource_account_configured` (bool): Set this to false if the bot is global.
+        The default is `true` for tennantized bots.
+        - `cloud` (str): Cloud environment that this identifier belongs to.
+
+    :param str bot_id: Microsoft bot id.
+    :keyword bool is_resource_account_configured: `False` if the identifier is global.
+    Default value is `True` for tennantzed bots.
+    :keyword cloud: Cloud environment that the bot belongs to. Default value is `PUBLIC`.
+    :paramtype cloud: str or ~azure.communication.chat.CommunicationCloudEnvironment
+    """
+    kind = CommunicationIdentifierKind.MICROSOFT_BOT
+
+    def __init__(self, bot_id: str, **kwargs: Any) -> None:
+        self.raw_id = kwargs.get('raw_id')
+        self.properties = MicrosoftBotProperties(
```

Comment:
You can remove the `#type` comment on line 254 in favour of this syntax:
```suggestion
        self.properties: MicrosoftBotProperties = MicrosoftBotProperties(
```

Reply:
Thanks for the comment! Updated!